java/src/main/java/com/logicalclocks/base/metadata/FeatureGroupBase.java

```diff
@@ -107,57 +108,41 @@ public class FeatureGroupBase {
   @Setter
   protected TimeTravelFormat timeTravelFormat = TimeTravelFormat.HUDI;
 
+  @Getter
+  @Setter
+  protected String onlineTopicName;
+
   @JsonIgnore
   protected Subject subject;
 
   protected FeatureGroupBaseEngine featureGroupBaseEngine = new FeatureGroupBaseEngine();
-  protected StatisticsEngine statisticsEngine = new StatisticsEngine(EntityEndpointType.FEATURE_GROUP);
   protected FeatureGroupUtils utils = new FeatureGroupUtils();
 
   private static final Logger LOGGER = LoggerFactory.getLogger(FeatureGroupBase.class);
 
-  public FeatureGroupBase(FeatureStore featureStore, Integer id) {
+  public FeatureGroupBase(FeatureStoreBase featureStore, Integer id) {
     this.featureStore = featureStore;
     this.id = id;
   }
 
-  public Query selectFeatures(List<Feature> features) {
-    return new Query(this, features);
-  }
-
-  public Query select(List<String> features) {
-    // Create a feature object for each string feature given by the user.
-    // For the query building each feature need only the name set.
-    List<Feature> featureObjList = features.stream().map(f -> new Feature(f, id)).collect(Collectors.toList());
-    return selectFeatures(featureObjList);
+  public QueryBase selectFeatures(List<Feature> features) {
+    return null;
   }
 
-  public Query selectAll() {
-    return selectAll(true, true);
+  public QueryBase select(List<String> features) {
+    return null;
   }
 
-  public Query selectAll(boolean includePrimaryKey, boolean includeEventTime) {
-    if (includePrimaryKey && includeEventTime) {
-      return new Query(this, getFeatures());
-    } else if (includePrimaryKey) {
-      return selectExcept(Lists.newArrayList(eventTime));
-    } else if (includeEventTime) {
-      return selectExcept(primaryKeys);
-    } else {
-      List<String> features = Lists.newArrayList(eventTime);
-      features.addAll(primaryKeys);
-      return selectExcept(features);
-    }
+  public QueryBase selectAll() {
+    return null;
   }
 
-  public Query selectExceptFeatures(List<Feature> features) {
-    List<String> exceptFeatures = features.stream().map(Feature::getName).collect(Collectors.toList());
-    return selectExcept(exceptFeatures);
+  public QueryBase selectExceptFeatures(List<Feature> features) {
+    return null;
```

Comment:
I think we still should define an abstract method for this in base class, since other frameworks would probably want this method. Furthermore, from the look of things, this (and other methods defined in `FeatureGroup` class) should be left in base class (even without defining it as abstract) since its functionality is not framework specific.